[ACR] `az acr create`: Fix domain name label suffix validation

trisavo-msft opened the pull request:
**Related command**
<!--- Please provide the related command with az {command} if you can, so that we can quickly route to the related person to review. --->

**Description**<!--Mandatory-->
<!--Why this PR? What is changed? What is the effect? etc. A high-quality description can accelerate the review process.-->

**Testing Guide**
<!--Example commands with explanations.-->

**History Notes**
<!--If your PR is not customer-facing, use {Component Name} in the PR title. Otherwise, use [Component Name] to allow our pipeline to add the title as a history note. If you need multiple history notes or would like to overwrite the note from the PR title, please fill in the following templates.-->

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

yonzhan commented:
Thank you for your contribution! We will review the pull request and get back to you soon.

nhu1997 commented on src/azure-cli/azure/cli/command_modules/acr/_validators.py:
```diff
@@ -120,7 +120,7 @@ def validate_registry_name(cmd, namespace):
     elif hasattr(suffixes, 'acr_login_server_endpoint'):
         acr_suffix = suffixes.acr_login_server_endpoint
-        pos = registry.find(acr_suffix)
+        pos = registry.ends_with(acr_suffix)
         if pos > 0:
```
Can you also fix the same issue with line #25731

zhoxing-ms commented:
/azp run

trisavo-msft commented:
> Can you document the reason for the removal of the test case? If you remove this test case, code coverage for handling registry names with a -dnlhash suffix in validate_registry_name will be lost and not covered by any other test in the repository?


@nhu1997  I spoke with Johnson Shi and confirmed that `az acr login -n acrname-dnlhash123` is not supposed to be supported. I have changed the validation method and the unit test accordingly. To make up for this lost test case, I have added this scenario in line 48 in the method `test_registry_name_invalid_inputs_should_raise_error`. `acrname-dnlhash123` should return an error instead of having the hash stripped. 

```
    def test_registry_name_invalid_inputs_should_raise_error(self):
        """Test invalid registry names that should raise InvalidArgumentValueError."""
        invalid_inputs = [
            "myregistry-hash123",  # Has hyphen but no suffix
```

zhoxing-ms commented:
/azp run
